src/rpc/client_channel.rs

```diff
@@ -236,24 +232,18 @@ impl WebRTCClientChannel {
             loop {
                 let split_at = MAX_REQUEST_MESSAGE_PACKET_DATA_SIZE
                     .min(data.len())
-                    .min(usize::try_from(next_message_length).unwrap());
+                    .min(next_message_length);
                 let (to_send, remaining) = data.split_at(split_at);
-                next_message_length -= u32::try_from(split_at).unwrap();
+                next_message_length -= split_at;
                 let stream = stream.clone();
+                let eos = remaining.is_empty();
                 let request = Request {
                     stream,
                     r#type: Some(Type::Message(RequestMessage {
                         has_message,
-                        eos: if !remaining.is_empty() {
-                            // stream definitely isn't done if there's more to send
-                            false
-                        } else {
-                            // if we intentionally sent an eos or the http request was inferrably
-                            // a stream
-                            eos || it_was_all_a_stream
-                        },
```

Comment:
Funny enough, we had a bit of a "two wrongs _do_ make a right" scenario here. We shouldn't be saying "stream done" every time we're in a stream (which we were doing before), but this actually led to client streaming working correctly instead because we were just defaulting to saying it was _always_ eos, which seemed to be fine.

The issue seems to be that in unary calls, `eos` (as far as I can tell) is meaningless so the fact that we were always saying "false" for eos in those cases (because `eos` was being passed as false, and because it definitely wasn't a stream so `it_was_all_a_stream` evaluated to false) didn't really matter. In client streaming cases, because we weren't ever re-evaluating the `it_was_all_a_stream` value, we were actually passing "true" here at all times! Thus, everything worked out okay.

 But in server streaming cases, `it_was_all_a_stream` was evaluating to false and so we were saying `eos: false`. This is exactly the same as in the unary case, which worked perfectly normally. My theory is that on the server side, `eos` is evaluated/matters in all streaming call cases (thus causing issues with our server streaming request despite the fact that the request is unary), but never in unary call cases (thus allowing unary requests to succeed despite the fact that `eos` was always false).